### PR TITLE
[Python] Fix 80 column violations

### PR DESCRIPTION
**Pull request by practicalswift**
#### What's in this pull request?

Fix 80 column violations in the project's Python code.
#### Resolved bug number: –

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: validation-test/stdlib/Slice/Inputs/GenerateSliceTests.py [hunk 1]
```diff
@@ -16,19 +16,37 @@
...
 //////// Do Not Edit Directly!
 // -*- swift -*-
-// RUN: rm -rf %t ; mkdir -p %t
-// RUN: %S/../../../utils/gyb %s -o %t/{testFilename} -D test_path="%S"
-// RUN: %S/../../../utils/line-directive %t/{testFilename} -- %target-build-swift %t/{testFilename} -o %t/{testFilename}.a.out
-// RUN: %S/../../../utils/line-directive %t/{testFilename} -- %target-run %t/{testFilename}.a.out
+// {run_directive_1}
+// {run_directive_2}
+// {run_directive_3}
+// {run_directive_4}
```
Comment (gribozavr) [inline validation-test/stdlib/Slice/Inputs/GenerateSliceTests.py]: I can't say that this part is an improvement...

File: utils/update-checkout [hunk 2]
```diff
@@ -74,12 +74,21 @@ def obtain_additional_swift_sources(
...
+                        ['git', 'clone', '--depth', '1', remote, dir_name])
                 else:
                     check_call(['git', 'clone', remote, dir_name])
                 if branch:
                     src_path = SWIFT_SOURCE_ROOT + "/" + dir_name + "/" + ".git"
-                    check_call(['git', '--git-dir', src_path, '--work-tree', os.path.join(SWIFT_SOURCE_ROOT, dir_name), 'checkout', branch])
+                    check_call([
+                        'git',
+                        '--git-dir',
+                        src_path,
```
Comment (gribozavr) [inline utils/update-checkout]: Can we put the command line flag and its argument on the same line?

File: test/Driver/Inputs/filelists/check-filelist-abc.py [hunk 3]
```diff
@@ -30,9 +30,12 @@
...
     lines = f.readlines()
-    assert lines[0].endswith("/a.swift\n") or lines[0].endswith("/a.swiftmodule\n")
-    assert lines[1].endswith("/b.swift\n") or lines[1].endswith("/b.swiftmodule\n")
-    assert lines[2].endswith("/c.swift\n") or lines[2].endswith("/c.swiftmodule\n")
+    assert lines[0].endswith(
+        "/a.swift\n") or lines[0].endswith("/a.swiftmodule\n")
+    assert lines[1].endswith(
+        "/b.swift\n") or lines[1].endswith("/b.swiftmodule\n")
+    assert lines[2].endswith(
+        "/c.swift\n") or lines[2].endswith("/c.swiftmodule\n")
```
Comment (gribozavr) [inline test/Driver/Inputs/filelists/check-filelist-abc.py]: This change feels like a regression in clarity.

File: benchmark/scripts/compare_perf_tests.py [hunk 4]
```diff
@@ -219,4 +225,5 @@ def usage():
         if key not in scores2:
             print(key, "not in", file2)
             continue
-        compare_scores(key, scores1[key], worstscores1[key], scores2[key], worstscores2[key], runs, nums[key])
+        compare_scores(key, scores1[key], worstscores1[key], scores2[
+                       key], worstscores2[key], runs, nums[key])
```
Comment (gribozavr) [inline benchmark/scripts/compare_perf_tests.py]: Breaking after `[` looks weird.

File: validation-test/stdlib/Slice/Inputs/GenerateSliceTests.py (hunk 1 above)
Comment (modocache) [inline validation-test/stdlib/Slice/Inputs/GenerateSliceTests.py]: I agree. Could we make an exception to apply this rule for lit RUN directives?

File: benchmark/scripts/compare_perf_tests.py (hunk 4 above)
Comment (modocache) [inline benchmark/scripts/compare_perf_tests.py]: I agree. Let's keep variables and their subscripts on the same line.

**Comment by modocache** [comment]
Looks fine to me, aside from @gribozavr's comments.

I imagine this will be one of the more controversial lint rules, but I'm glad we will have a rule in place at all.

File: validation-test/stdlib/Slice/Inputs/GenerateSliceTests.py (hunk 1 above)
Comment (practicalswift) [inline validation-test/stdlib/Slice/Inputs/GenerateSliceTests.py]: @gribozavr I totally agree - it is really really ugly :-)

Short of doing an ugly work around I think the best solution would be to simply skip linting of this file using the "# flake8: noqa" directive.

**Comment by practicalswift** [comment]
@gribozavr @modocache Thanks for reviewing and thanks for good input.

I've tried to address the issues identified by @gribozavr - please let me know if you find any additional things that needs fixing.

As for the ugly work around in `validation-test/stdlib/Slice/Inputs/GenerateSliceTests.py` I removed that and excluded that file from linting (using `# flake8: noqa`). There simply does not seem to exist a way to keep that file under 80 columns without sacrificing readability :-)

**Comment by gribozavr** [comment]
@practicalswift Thank you, I don't have further comments.

**Comment by rintaro** [comment]
@practicalswift 
Python support `\` escape even in `"""`. So:
```
"""
// RUN: %S/../../../utils/gyb %s \
    -o %t/{testFilename} -D test_path="%S"
"""
```
should works.

OR, `lit.py` supports `\` escape in `RUN:` [like this](https://github.com/apple/swift/blob/0796eaad1fa99eae91460956c4ad28f2c047be0a/test/Misc/tbi.swift), So:
```
"""
// RUN: %S/../../../utils/gyb %s \\
// RUN:     -o %t/{testFilename} -D test_path="%S"
"""
```
should also works.

**Comment by practicalswift** [comment]
@rintaro Oh, I wasn't aware of that. Thanks a lot for noticing! :+1: 

I'll submit a PR fixing this right away :-)

**Pull request merged**